extensions/vscode/src/api/client.ts

```diff
@@ -77,10 +95,11 @@ let api: PublishingClientApi | undefined = undefined;
 // NOTE: this function must be called ahead of useApi()
 // so that the class is properly instantiated.
 export const initApi = (
+  insecureSettingFn: GetInsecureSettingFn,
```

Comment:
It is not clear to me why this is part of the API client.

Why not keep the parameters for the API calls that care about the insecure setting, and use a helper function in the extension to determine what to send to those functions like you are already doing?

```
workspace.getConfiguration("positPublisher").get<boolean>("enableInsecureTLS")
```

Reply:
As discussed, I'll move the helper functionality out.